RFS/src/main/java/org/opensearch/migrations/bulkload/common/OpenSearchClient.java

```diff
@@ -381,8 +381,10 @@ protected Retry getBulkRetryStrategy() {
     public Mono<BulkResponse> sendBulkRequest(String indexName, List<BulkDocSection> docs,
                                               IRfsContexts.IRequestContext context)
     {
+        final AtomicInteger attemptCounter = new AtomicInteger(0);
         final var docsMap = docs.stream().collect(Collectors.toMap(d -> d.getDocId(), d -> d));
         return Mono.defer(() -> {
+            int attempt = attemptCounter.incrementAndGet();
```

Comment:
Nit: we don't need this local, get the counter during the message construction.
```suggestion
            attemptCounter.incrementAndGet();
```